silk-react-components/src/HierarchicalMapping/Components/MappingRule/SuggestionsView.jsx

```diff
@@ -40,106 +34,123 @@ const SuggestionsView = React.createClass({
-                        },
-                        (err) => {
-                            console.warn('err MappingRuleOverview: rule.suggestions');
-                            this.setState({loading: false});
-                        },
-                    );
-                },
-                (err) => {
-                    console.warn('err MappingRuleOverview: transform.get');
-                    this.setState({loading: false});
+    loadData(){
+        this.setState({
+            loading: true,
+        });
+        pendingRules = {};
+        wrongRules = {}
+        hierarchicalMappingChannel.request(
+            {
+                topic: 'rule.suggestions',
+                data: {
+                    targetClassUris: this.props.targetClassUris,
+                    ruleId: this.props.ruleId,
                 }
-            )
+            }).subscribe(
+            (response) => {
+                this.setState({
+                    loading: false,
+                    data: response.suggestions,
+                });
+            },
+            (err) => {
+                console.warn('err MappingRuleOverview: rule.suggestions');
+                this.setState({loading: false});
+            },
+        );
+
+    },
+    componentDidMount() {
+        this.loadData();
     },
     handleAddSuggestions(event) {
         event.stopPropagation();
-        let remaining = this.state.checked.length;
+        let correspondences = [];
         this.setState({
             loading: true,
         })
         _.map(this.state.checked, (suggestion) => {
             const a = _.split(suggestion, ';');
-            const suggestionForAdding = this.state.data[a[0]][a[1]];
-            suggestionForAdding.targetClassUri = a[0];
-            // FIXME: atomize it with only 1 call to DI or stop when the first save fails?
-            hierarchicalMappingChannel.request({
-                topic: 'rule.createValueMapping',
-                data: {
-                    id: undefined,
-                    parentId: this.props.id,
-                    type: 'direct',
-                    comment: '',
-                    targetProperty: suggestionForAdding.targetClassUri,
-                    propertyType: '',
-                    sourceProperty: suggestionForAdding.uri,
-                }
-            }).subscribe(
-                () => {
-                    remaining--;
-                    if (remaining === 0){
-                        hierarchicalMappingChannel.subject('reload').onNext(true);
-                        this.setState({loading: false});
-                        this.props.onClose(event);
-                    }
-
-                }, (err) => {
-                    this.setState({
-                        error: err,
-                        loading: false,
-                    });
-                });
-
-
+            correspondences.push({
+                sourcePath: this.state.data[a[0]][a[1]].uri,
```

Comment:
done: {suggestedClass, pos} is the new definition, instead of `${suggestedClass};${pos}`
use _.some instead of _include required to check if contained